error: Could not compute dependency graph: Error Domain=NSPOSIXErrorDomain Code=86 "Bad CPU type in executable"

When `swift-build` computes the dependency graph for a build it seems to resolve the toolchains available on the system, which may include tools that are `x86_64` only. If Rosetta2 is not installed, this results in a hard-failure for every build in Xcode/xcodebuild on Apple Silicon system:

<img width="1024" height="708" alt="Image" src="https://github.com/user-attachments/assets/bd1c7961-e215-4430-b4e9-2e6d1c9cd0b2" />

Installing Rosetta2, and then watching process creation via [Process Monitor](https://objective-see.org/products/utilities.html#ProcessMonitor) during a build shows:

```json
{
  "event": "ES_EVENT_TYPE_NOTIFY_EXEC",
  "timestamp": "2025-10-02 09:39:14 +0000",
  "process": {
    "pid": 35254,
    "name": "qcc",
    "path": "/opt/qnx700/host/darwin/x86_64/usr/bin/qcc",
    "uid": 501,
    "architecture": "unknown",
    "arguments": [
      "/opt/qnx700/host/darwin/x86_64/usr/bin/qcc",
      "-dM",
      "E",
      "-x",
      "c",
      "-c",
      "/dev/null",
      "-o",
      "/dev/null"
    ],
    "ppid": 35253,
    "rpid": 34691,
    "ancestors": [
      34691,
      1
    ],
    "signing info (reported)": {
      "csFlags": 0,
      "platformBinary": 0,
      "signingID": "",
      "teamID": "",
      "cdHash": "0000000000000000000000000000000000000000"
    },
    "signing info (computed)": {
      "signatureStatus": -67062
    }
  }
}
```

With `SWBBuildService` as its parent process.

I don't know if there's any way to opt out of (specific) toolchain lookups? 

In any case, the logic [here for QNX](https://github.com/swiftlang/swift-build/blob/823f4f8c84bbf8e6a6752d4550fafaebfec43bb3/Sources/SWBQNXPlatform/QNXSDP.swift#L41), or possibly at a higher level, should probably handle the situation gracefully by treating it as the toolchain not found, possibly with an explicit warning, rather than the opaque "Bad CPU type in executable" with no info about _which_ executable failed to run 😄

As a data point, renaming the `qcc` binary to `qcc.nope` results in:
```
$ xcodebuild
Command line invocation:
    /Applications/Xcode26_Universal.app/Contents/Developer/usr/bin/xcodebuild

ComputePackagePrebuildTargetDependencyGraph
error: Could not compute dependency graph: stat(/opt/qnx700/host/darwin/x86_64/usr/bin/qcc): No such file or directory (2)

** BUILD FAILED **


The following build commands failed:
	Compute target dependency graph for package preparation
(1 failure)
```

While renaming `/opt/qnx700` results in a successful build.

--- a/Sources/SWBQNXPlatform/Plugin.swift
+++ b/Sources/SWBQNXPlatform/Plugin.swift
@@ -45,7 +45,7 @@ struct QNXEnvironmentExtension: EnvironmentExtension {
     let plugin: QNXPlugin
 
     func additionalEnvironmentVariables(context: any EnvironmentExtensionAdditionalEnvironmentVariablesContext) async throws -> [String : String] {
-        if let latest = try await plugin.cachedQNXSDPInstallations(host: context.hostOperatingSystem).first {
+        if let latest = try? await plugin.cachedQNXSDPInstallations(host: context.hostOperatingSystem).first {
             return .init(latest.environment)
         }
         return [:]

